cli/cmd/diagnose.go

```diff
@@ -0,0 +1,148 @@
+package cmd
+
+import (
+	"compress/gzip"
+	"context"
+	"fmt"
+	"github.com/odigos-io/odigos/cli/cmd/resources"
+	"github.com/odigos-io/odigos/cli/pkg/kube"
+	"github.com/spf13/cobra"
+	"io"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"os"
+	"path/filepath"
+)
+
+const (
+	mainDir       = "odigos-diagnose"
+	logDir        = "logs"
+	logBufferSize = 1024 * 1024 // 1MB buffer size for reading logs in chunks
+)
+
+var diagnozeCmd = &cobra.Command{
+	Use:   "diagnose",
+	Short: "Diagnose Client Cluster",
+	Long:  `Diagnose Client Cluster to identify issues and resolve them. This command is useful for troubleshooting and debugging.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := cmd.Context()
+		client, err := kube.CreateClient(cmd)
+		if err != nil {
+			kube.PrintClientErrorAndExit(err)
+		}
+
+		err = startDiagnose(ctx, client)
+		if err != nil {
+			fmt.Printf("The diagnose script crashed on: %v\n", err)
+		}
+	},
+}
+
+func startDiagnose(ctx context.Context, client *kube.Client) error {
+	if err := createAllDirs(); err != nil {
+		return err
+	}
+
+	if err := fetchOdigosComponentsLogs(ctx, client); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createAllDirs() error {
+	if err := os.RemoveAll(mainDir); err != nil {
+		return err
+	}
+	if err := os.MkdirAll(mainDir, 0755); err != nil {
```

Comment:
I think this will create the directory in the same place where `odigos diagnose` is run. Maybe better to always create it at some temp directory. I think Go has a built in function like `ioutil.TempDir` or something similar

Reply:
Why don't we want it to be saved where the user run  odigos diagnose?